core/src/snapshot_package.rs

```diff
@@ -0,0 +1,219 @@
+use crate::result::{Error, Result};
+use crate::service::Service;
+use crate::temp_files::TempFiles;
+use flate2::write::GzEncoder;
+use flate2::Compression;
+use solana_runtime::accounts_db::AccountStorageEntry;
+use std::fs;
+use std::fs::File;
+use std::path::Path;
+use std::path::PathBuf;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::mpsc::{Receiver, RecvTimeoutError, Sender};
+use std::sync::Arc;
+use std::thread::{self, Builder, JoinHandle};
+use std::time::Duration;
+
+pub type SnapshotPackageSender = Sender<SnapshotPackage>;
+pub type SnapshotPackageReceiver = Receiver<SnapshotPackage>;
+
+pub struct SnapshotPackage {
+    snapshot_path: PathBuf,
+    storage_entries: Vec<Arc<AccountStorageEntry>>,
+}
+
+impl SnapshotPackage {
+    pub fn new(snapshot_path: PathBuf, storage_entries: Vec<Arc<AccountStorageEntry>>) -> Self {
+        Self {
+            snapshot_path,
+            storage_entries,
+        }
+    }
+}
+
+impl Drop for SnapshotPackage {
+    fn drop(&mut self) {
+        let _ = fs::remove_dir_all(&self.snapshot_path);
+    }
+}
+
+pub struct SnapshotPackagerService {
+    t_snapshot_packager: JoinHandle<()>,
+}
+
+impl SnapshotPackagerService {
+    pub fn new(
+        snapshot_package_receiver: SnapshotPackageReceiver,
+        exit: &Arc<AtomicBool>,
+        snapshot_package_output_path: String,
+    ) -> Self {
+        let exit = exit.clone();
+        let snapshot_package_output_path = PathBuf::from(snapshot_package_output_path);
+        let t_snapshot_packager = Builder::new()
+            .name("solana-snapshot-packager".to_string())
+            .spawn(move || loop {
+                if exit.load(Ordering::Relaxed) {
+                    break;
+                }
+                if let Err(e) = Self::package_snapshots(
+                    &snapshot_package_receiver,
+                    &snapshot_package_output_path,
+                ) {
+                    match e {
+                        Error::RecvTimeoutError(RecvTimeoutError::Disconnected) => break,
+                        Error::RecvTimeoutError(RecvTimeoutError::Timeout) => (),
+                        _ => info!("Error from package_snapshots: {:?}", e),
+                    }
+                }
+            })
+            .unwrap();
+        Self {
+            t_snapshot_packager,
+        }
+    }
+
+    pub fn package_snapshots(
+        snapshot_receiver: &SnapshotPackageReceiver,
```

Comment:
once inside a module, consider dropping the module name as a prefix for variables